client/my-sites/themes/trending-themes.tsx

```diff
@@ -5,16 +5,25 @@ import {
 	getTrendingThemes as getTrendingThemesSelector,
 	areTrendingThemesLoading,
 } from 'calypso/state/themes/selectors';
+import { TrendingTheme, TrendingThemesFilter } from 'calypso/types';
 import { ConnectedThemesSelection } from './themes-selection';
 
-class TrendingThemes extends Component {
+interface TrendingThemesProps {
+	customizedThemesList: TrendingTheme[];
+	filter: TrendingThemesFilter;
+	getTrendingThemes: ( filter: TrendingThemesFilter ) => Promise< void >;
+	isLoading: boolean;
+	scrollToSearchInput: () => void;
+}
+
+class TrendingThemes extends Component< TrendingThemesProps > {
 	componentDidMount() {
 		if ( ! this.props.customizedThemesList.length ) {
 			this.fetchThemes();
 		}
 	}
 
-	componentDidUpdate( prevProps ) {
+	componentDidUpdate( prevProps: { isLoading: boolean; filter: TrendingThemesFilter } ) {
```

Comment:
That's what I thought as well, but for some reason, typescript was complaining in my local environment. 🤔

Reply:
Oh, wow! That's true indeed! `React.Component` does not include the lifecycle methods 🤔 

Well, in this case, instead of inlining the type, we might put `TrendingThemesProps`. Like:

```ts
prevProps: TrendingThemesProps
```

This way we can:
1.  be consistent regarding declarations;
2. define a single point of maintenance.

Reply:
Makes total sense. Done 🙂